packages/engine/Source/Shaders/Model/FogStageFS.glsl

```diff
@@ -0,0 +1,94 @@
+vec3 computeEllipsoidPosition(vec3 positionMC) {
```

Comment:
Most of this code is identical to code in vertical exaggeration. Consider pulling out a builtin function for this.

The one tricky thing is vertical exaggeration only needs the height, but fog needs the position on the ellipsoid in world coordinates. So maybe split it into smaller functions.

Reply:
I ended up not using this version of the function, so moot point.